test/integration_test/storkctl_schedulepolicy_test.go

```diff
@@ -0,0 +1,196 @@
+	createSchedulePolicyTest(t, "Daily", nil, "daily-policy.yaml")
+}
+
+func createDefaultWeeklySchedulePolicyTest(t *testing.T) {
+	createSchedulePolicyTest(t, "Weekly", nil, "weekly-policy.yaml")
+}
+
+func createDefaultMonthlySchedulePolicyTest(t *testing.T) {
+	createSchedulePolicyTest(t, "Monthly", nil, "monthly-policy.yaml")
+}
+
+func createCustomIntervalSchedulePolicyTest(t *testing.T) {
+	args := map[string]string{"interval-minutes": "15", "retain": "5"}
+	createSchedulePolicyTest(t, "Interval", args, "custom-interval-policy.yaml")
+}
+
+func createCustomDailySchedulePolicyTest(t *testing.T) {
+	args := map[string]string{"force-full-snapshot-day": "Wednesday", "time": "4:00PM", "retain": "2"}
+	createSchedulePolicyTest(t, "Daily", args, "custom-daily-policy.yaml")
+}
+
+func createCustomWeeklySchedulePolicyTest(t *testing.T) {
+	args := map[string]string{"day-of-week": "Friday", "time": "2:00AM", "retain": "3"}
+	createSchedulePolicyTest(t, "Weekly", args, "custom-weekly-policy.yaml")
+}
+
+func createCustomMonthlySchedulePolicyTest(t *testing.T) {
+	args := map[string]string{"date-of-month": "15", "time": "11:00AM", "retain": "3"}
+	createSchedulePolicyTest(t, "Monthly", args, "custom-monthly-policy.yaml")
+}
+
+func ValidateSchedulePolicyFromFile(t *testing.T, specFile string, policyName string) error {
+	if specFile == "" {
+		return fmt.Errorf("empty file path")
+	}
+	file, err := os.Open(specFile)
+	if err != nil {
+		return fmt.Errorf("error opening file %v: %v", specFile, err)
+	}
+	data, err := io.ReadAll(bufio.NewReader(file))
+	if err != nil {
+		return fmt.Errorf("error reading file %v: %v", specFile, err)
+	}
+	policy := &storkv1.SchedulePolicy{}
+	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(data)), len(data))
+	if err := dec.Decode(&policy); err != nil {
+		return err
+	}
+
+	if err := ValidateSchedulePolicy(t, policy, policyName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidateSchedulePolicy(t *testing.T, schedulePolicy *storkv1.SchedulePolicy, schedulePolicyName string) error {
+	//We want to validate if the created schedule policy resource matches our expectations
+	actualPolicy, err := storkops.Instance().GetSchedulePolicy(schedulePolicyName)
+	if err != nil {
+		logrus.Errorf("unable to get schedule policy")
+		return err
+	}
+	logrus.Info("Trying to validate if the created policy is per expectations")
+	//Validating schedulePolicy.Policy because schedulePolicy.metadata cannot be validated
+	require.Equal(t, schedulePolicy.Policy, actualPolicy.Policy, "Created schedule policy doesn't match the expected specification provided")
+	return nil
+}
+
+func SchedulePolicyCleanup(t *testing.T, policyName string) {
+	fmt.Println("Cleaning up created resources")
+	err := storkops.Instance().DeleteSchedulePolicy(policyName)
+	if err != nil {
+		logrus.Errorf("unable to delete schedule policy %v", policyName)
+	}
+	// time to let deletion finish
+	time.Sleep(time.Second * 2)
```

Comment:
I think we can explicitly call SchedulePolicyCleanup, call delete and wait for it to get deleted instead of putting a random sleep here.

Reply:
Thanks for the suggestion. I realise that the sleep here is redundant.

Reply:
Have used schedops' task.DoRetryWithTimeout to check if schedule policy is successfully deleted as part of the SchedulePolicyCleanup function.